src/container.cpp

```diff
@@ -353,18 +351,18 @@ ReturnValue Container::queryMaxCount(int32_t index, const Thing& thing, uint32_t
 		if (index == INDEX_WHEREEVER) {
 			// Iterate through every item and check how much free stackable slots there is.
 			uint32_t slotIndex = 0;
-			for (Item* containerItem : itemlist) {
-				if (containerItem != item && containerItem->equals(item) &&
+			for (const auto& containerItem : itemList) {
+				if (containerItem != item && *containerItem == *item &&
 				    containerItem->getItemCount() < ITEM_STACK_SIZE) {
-					if (queryAdd(slotIndex++, *item, count, flags) == RETURNVALUE_NOERROR) {
+					if (queryAdd(slotIndex++, item, count, flags) == RETURNVALUE_NOERROR) {
 						n += ITEM_STACK_SIZE - containerItem->getItemCount();
 					}
 				}
 			}
 		} else {
-			const Item* destItem = getItemByIndex(index);
-			if (item->equals(destItem) && destItem->getItemCount() < ITEM_STACK_SIZE) {
-				if (queryAdd(index, *item, count, flags) == RETURNVALUE_NOERROR) {
+			if (auto destItem = getItemByIndex(index);
```

Comment:
Could this be `const auto`
```suggestion
			if (const auto destItem = getItemByIndex(index);
```